pkg/registry/core/componentstatus/rest.go

```diff
@@ -74,105 +69,12 @@ func (rs *REST) NewList() runtime.Object {
 	return &api.ComponentStatusList{}
 }
 
-// Returns the list of component status. Note that the label and field are both ignored.
-// Note that this call doesn't support labels or selectors.
 func (rs *REST) List(ctx context.Context, options *metainternalversion.ListOptions) (runtime.Object, error) {
-	servers := rs.GetServersToValidate()
-
-	wait := sync.WaitGroup{}
-	wait.Add(len(servers))
-	statuses := make(chan api.ComponentStatus, len(servers))
-	for k, v := range servers {
-		go func(name string, server *Server) {
-			defer wait.Done()
-			status := rs.getComponentStatus(name, server)
-			statuses <- *status
-		}(k, v)
-	}
-	wait.Wait()
-	close(statuses)
-
-	pred := componentStatusPredicate(options)
-
-	reply := []api.ComponentStatus{}
-	for status := range statuses {
-		// ComponentStatus resources currently (v1.14) do not support labeling, however the filtering is executed
-		// nonetheless in case the request contains Label or Field selectors (which will effectively filter out
-		// all of the results and return an empty response).
-		if matched := matchesPredicate(status, &pred); matched {
-			reply = append(reply, status)
-		}
-	}
-	return &api.ComponentStatusList{Items: reply}, nil
-}
-
-func componentStatusPredicate(options *metainternalversion.ListOptions) storage.SelectionPredicate {
-	pred := storage.SelectionPredicate{
-		Label:    labels.Everything(),
-		Field:    fields.Everything(),
-		GetAttrs: nil,
-	}
-	if options != nil {
-		if options.LabelSelector != nil {
-			pred.Label = options.LabelSelector
-		}
-		if options.FieldSelector != nil {
-			pred.Field = options.FieldSelector
-		}
-	}
-	return pred
-}
-
-func matchesPredicate(status api.ComponentStatus, pred *storage.SelectionPredicate) bool {
-	// currently no fields except the generic meta fields are supported for predicate matching
-	fieldsSet := generic.AddObjectMetaFieldsSet(make(fields.Set, 2), &status.ObjectMeta, true)
-	return pred.MatchesObjectAttributes(
-		status.ObjectMeta.Labels,
-		fieldsSet,
-	)
+	return &api.ComponentStatusList{Items: []api.ComponentStatus{}}, nil
```

Comment:
I disagree with this change.

API deprecation means that we won't be evolving it.
But GA APIs (and componentstatuses kind-of implicitly belongs to this category) can't be broken.
Not exposing this information has a potential to break some users and I'm definitely against doing this.

@thockin @deads2k - in absence of Jordan.

Reply:
cc @logicalhan who proposed component health SLIs which can be used for replacement.

Reply:
The information is wrong, the assumption about the probes addresses and ports can not be generalized, assuming that controller manager and scheduler are reachable through localhost is not always true

Reply:
I know that, but there might be users for who it's true and depend on it.

Reply:
I'm against this change. I realize this is API compliant by the letter of the law, but this very much violates the spirit of the law. 

> cc @logicalhan who proposed component health SLIs which can be used for replacement.

Component-SLIs won't be complete until we have those metrics in etcd as well. And this endpoint surfaces etcd health statuses, which we currently don't have a replacement for (except probing the etcd `/health` endpoint directly).